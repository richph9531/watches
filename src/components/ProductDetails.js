import React from 'react';
import PropTypes from 'prop-types';

import ProductHeader from './ProductHeader';

import Features from './Features';

import classes from '../styles/ProductCard.module.css';

const ProductDetails = props => {
  const ProductCards = props.data.colorOptions.map((item, pos) => (
    <img
      key={pos}
      src={item.imageUrl}
      alt={item.styleName}
      className={props.isCurrentlySelected ? classes.ProductCardSelected : classes.ProductCard }
      onClick={() => props.onProductSelection(pos)}
    />
  ));

  return (
    <div>
      <div><ProductHeader data={props.data}/></div>
      <div>
        <h3>Select Colour</h3>
        <div>{ ProductCards }</div>
      </div>
      < Features data={props}/>
    </div>
  );
};

ProductDetails.propTypes = {
  data: PropTypes.object,
  imageUrl: PropTypes.string,
  styleName: PropTypes.string,
  onProductSelection: PropTypes.func,
  isCurrentlySelected: PropTypes.bool,
};

export default ProductDetails;
