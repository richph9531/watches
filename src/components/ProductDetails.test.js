import React from 'react';

import { render } from '@testing-library/react';
import ProductDetails from './ProductDetails';

test.skip('Component renders correctly with Time selected', () => {
  const data = {
    data: {
      colorOptions: [
        {
          id: 1,
          styleName: 'Black Strap',
          imageUrl: 'BlackStrap.png',
        },
        {
          id: 2,
          styleName: 'Red Strap',
          imageUrl: 'RedStrap.png',
        },
      ],
    },
  };
  const renderedElement = render(<ProductDetails data={data}/>);
  expect(renderedElement.baseElement).toMatchSnapshot();
});

