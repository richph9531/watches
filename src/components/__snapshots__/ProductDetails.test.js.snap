// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly with Time selected 1`] = `
<body>
  <div>
    <div>
      <h3
        class="FeaturesHeading"
      >
        Features
      </h3>
      <div>
        <button
          class="SelectedFeatureItem"
        >
          Time
        </button>
        <button
          class="FeatureItem"
        >
          Heart Rate
        </button>
      </div>
      <div>
        <button
          class="BuyNow"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</body>
`;
